### DISCUSSION TITLE
[INI configuration file] how to update a variable from an inherited preset?

### DISCUSSION THREAD

**Discussion by HitLuca** [supermerill/SuperSlicer#3180] [Q&A]
I found myself in the need of updating an existing entry for a filament profile, more specifically, have a custom variable defined in the parent profile, and a second custom variable added in the children profile. An example is below:

```
[filament: Some PLA]
inherits_from=*PLA*
filament_custom_variables=filament_name="Some PLA"

[filament: *PLA*]
filament_custom_variables=pressure_advance=0.123
```

You can see above how I want to end up with both `filament_name` and `pressure_advance` as defined variables when I use the `Some PLA` filament, but the `pressure_advance` one gets overwritten when the inheriting profile gets loaded.

How can I solve this?

**Comment by supermerill** [comment]
currently, you can't, you have to rewrite it entirely
```
[filament: Some PLA]
inherits = *PLA*
filament_custom_variables = pressure_advance=0.123\nfilament_name="Some PLA"
```

I guess it may be possible to kind of hack the parser to make it parse '= +=' like "append" but it's really not trivial.

**Reply by HitLuca** [reply]
yes, having something like `+=` would be the solution to this issue, but I guess it's niche enough that it won't be seen around soon. I sadly don't have the time to check out the codebase and make the changes myself, but I will see if I can come up with an alternative that doesn't involve repeating values on every filament I have.

Since I figured out how to get the filament name from settings (`[filament_settings_id[0]]`) it's not as important now, but still something that can be looked into
